### PR TITLE
Add the Redis secrets engine

### PR DESCRIPTION
**Pull request by remilapeyre**
Closes https://github.com/hashicorp/vault/issues/11697

### CODE REVIEW

**Comment by hsimon-hashicorp** [comment]
Hi again @remilapeyre - this one needs a changelog too, if you would please. :)

**Comment by remilapeyre** [comment]
Thanks fr the ping @hsimon-hashicorp

File: website/content/api-docs/secret/redis.mdx [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+layout: api
+page_title: Redis - Secrets Engines - HTTP API
+description: This is the API documentation for the Vault Redis secrets engine.
+---
+
+# Redis Secrets Engine (API)
+
+This is the API documentation for the Vault Redis secrets engine. For general
+information about the usage and operation of the Redis secrets engine, please
+see the [Redis secret engine documentation](/docs/secrets/redis).
```
Comment (taoism4504) [inline website/content/api-docs/secret/redis.mdx:11]:
```suggestion
see the [Redis secrets engine documentation](/docs/secrets/redis).
```

File: website/content/api-docs/secret/redis.mdx [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+### Parameters
+
+- `address` `(string: <required>)` - Specifies the address of the Redis server.
+
+- `username` `(string: <required>)` - Specifies the Redis user to connect with.
+
+- `password` `(string)` - Specifies the password to use.
+
+- `rotate` `(bool)` - Whether to rotate the password immediately. This ensures
```
Comment (taoism4504) [inline website/content/api-docs/secret/redis.mdx:33]:
```suggestion
- `rotate` `(bool)` - Specifies whether to rotate the password immediately. This ensures
```

File: website/content/api-docs/secret/redis.mdx [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+  that only Vault knows the password of the administrative user, it is
```
Comment (taoism4504) [inline website/content/api-docs/secret/redis.mdx:34]:
```suggestion
  that only Vault knows the password of the administrative user. It is
```

File: website/content/api-docs/secret/redis.mdx [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+    password="password" \
+    rotate=true
+```
+
+</Tab>
+</Tabs>
+
+## Rotate Root Credentials
+
+This endpoint rotates the password used by Vault. Once this method is called
```
Comment (taoism4504) [inline website/content/api-docs/secret/redis.mdx:77]:
```suggestion
This endpoint rotates the password used by Vault. Once this method is called,
```

**Comment by hsimon-hashicorp** [comment]
Hi @remilapeyre - I wanted to come back to this and let you know that we are discussing this internally. This is a little bit more involved, hence the delay. We're aware of the level of effort you've spent on this PR, and I wanted to make sure we communicated that it's appreciated, and our engineering team is talking about it! :)

**Comment by remilapeyre** [comment]
Thanks for letting me know. To give more context this is a secret engine that was asked to us by a Vault Enterprise client. Since it is also one of the most asked for by the community I think it makes more sense to integrate it in Vault rather than distributing it as a separate plugin: this makes it easier as Redis is a very common infrastructure piece and avoid having to audit this and update a separate binary. I think this could be useful to many Vault users. 

I know that an additional secret engine also adds maintenance so not all secret engines can be a part of Vault core but the Redis protocol and feature set is very stable so I think it would be very manageable. I also took the liberty to add myself to the CODEOWNERS file since I don’t mind helping if there is any issue to fix or updates to do. Let me know if you think this is an issue and I should remove myself from there.

**Comment by Zlaticanin** [comment]
Hello Remi, thank you for your contribution. We decided not to proceed with this solution because we want to integrate Redis as a DB plugin, as opposed to the general secrets plugin. We preferred Redis to be a DB plugin because it conforms to a common Database interface, making it easier to maintain and adding less cognitive load to users who are already familiar with the Database secrets engine. During discovery, we found a community repository that supports any kind of Redis OSS installation and we decided to proceed with that one instead. We talked to the contributor and we asked if we could take ownership of his repository, to which he agreed. The plugin is now part of HashiCorp, and has been tested to be fully functional. We are still working on some small modifications and creating the docs, but the plugin will be available on the next major Vault release. Thank you!

**Comment by Zlaticanin** [comment]
https://github.com/hashicorp/vault-plugin-database-redis

**Pull request closed without merging**